pkg/gateway/server/oauth_apps.go

```diff
@@ -125,6 +126,20 @@ func (s *Server) createOAuthApp(apiContext api.Context) error {
 		return types2.NewErrHTTP(http.StatusConflict, fmt.Sprintf("OAuth app with alias %s already exists", appManifest.Alias))
 	}
 
+	// Store the client secret as a credential.
+	credential := gptscript.Credential{
+		Context:  fmt.Sprintf("%s-%s", appManifest.Alias, appManifest.ClientID),
+		ToolName: appManifest.Alias,
+		Type:     gptscript.CredentialTypeTool,
+		Env:      map[string]string{"client_secret": appManifest.ClientSecret},
```

Comment:
nit: `Env` here means `CLIENT_SECRET` to me.

Reply:
Also, what are your thoughts on the following?
1. Create the OAuth app first.
2. Put the credential in the credential context corresponding to the generated name of the OAuth app.

My thoughts on this are:
1. Allows the same OAuth credentials to be used for multiple OAuth apps (assuming we support that one day).
2. Allows for multiple OAuth apps with the same `alias` (assuming we support that one day).

Reply:
I can update that string. It's not actually an environment variable, since we are using creds outside the context of tools here. So it doesn't really make a difference whether it's caps or not, as long as we use the same string later when we read the credential.

Regarding this:
> Allows the same OAuth credentials to be used for multiple OAuth apps (assuming we support that one day).

I'm not sure I understand how this would allow for that.

Reply:
This implementation doesn't allow for the same Client ID to be used for the same alias: when you try to create a second OAuth app with the same client credentials, you will get an error each time about the credential already existing.

If you put these credentials in a credential context corresponding to the generated name of the OAuth app, then this type of collision won't occur.

Reply:
Oh now I understand. Thanks.

Reply:
Fixed